base/process.jl

```diff
@@ -72,6 +72,13 @@ backticks, e.g.
 """
 Cmd
 
+hash(x::Cmd) = hash(x.exec, hash(x.env, hash(x.ignorestatus, hash(x.dir, hash(x.flags)))))
+function ==(x::Cmd, y::Cmd)
+    a = isequal(x.exec, y.exec) && isequal(x.env, y.env)
```

Comment:
I think that's the preferred style, since there aren't any other options (other than creating temporary variables, which isn't great).